### PR TITLE
Add OP & CBW Quests to Labels

### PR DESCRIPTION
**Pull request by MSilb7**
Brief comments on the purpose of your changes:
- Create spells for Coinbase Wallet & OP Quests
- Add participants to labels
- Add labels for Optimism Quest tiers

**For Dune Engine V2**

I've checked that:

### General checks:
* [ ] I tested the query on dune.com after compiling the model with dbt compile (compiled queries are written to the target directory)
* [ ] I used "refs" to reference other models in this repo and "sources" to reference raw or decoded tables 
* [ ] if adding a new model, I added a test
* [ ] the filename is unique and ends with .sql
* [ ] each sql file is a select statement and has only one view, table or function defined  
* [ ] column names are `lowercase_snake_cased`
* [ ] if adding a new model, I edited the dbt project YAML file with new directory path for both models and seeds (if applicable)
* [ ] if wanting to expose a model in the UI (Dune data explorer), I added a post-hook in the JINJA config to add metadata (blockchains, sector/project, name and contributor Dune usernames)

### Pricing checks:
* [ ] `coin_id` represents the ID of the coin on coinpaprika.com
* [ ] all the coins are active on coinpaprika.com (please remove inactive ones)

### Join logic:
* [ ] if joining to base table (i.e. ethereum transactions or traces), I looked to make it an inner join if possible

### Incremental logic:
* [ ] I used is_incremental & not is_incremental jinja block filters on both base tables and decoded tables
  * [ ] where block_time >= date_trunc("day", now() - interval '1 week')
* [ ] if joining to base table (i.e. ethereum transactions or traces), I applied join condition where block_time >= date_trunc("day", now() - interval '1 week')
* [ ] if joining to prices view, I applied join condition where minute >= date_trunc("day", now() - interval '1 week')

### CODE REVIEW

**Comment by MSilb7** [comment]
Getting an error on uniques at the labels table level, but not getting it in any of the intermediate queries

**Comment by Hosuke** [comment]
> Getting an error on uniques at the labels table level, but not getting it in any of the intermediate queries

Duped records details:
https://dune.com/queries/2203547?d=1

It seems from two same models/labels in `persona` and `usage`.

File: models/coinbase_wallet_quests/optimism/coinbase_wallet_quests_optimism_rewards_transfers.sql [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+WHERE evt_block_time >= cast('{{project_start_date}}' as timestamp) --arbitrary
+{% if is_incremental() %}
+-- for quest addresses we've seen before, pull incremental, else pull everything (controls for if we first see a distributor address later)
+AND 1 = (
+        CASE WHEN evt_block_time >= date_trunc('day', now() - interval '1 week') THEN 1
+             WHEN distributor_address NOT IN (SELECT distributor_address FROM {{this}} GROUP BY 1) THEN 1--we don't have this loaded in yet.
+             ELSE 0
+        END
+        )
+{% endif %}
```
Comment (Hosuke) [inline models/coinbase_wallet_quests/optimism/coinbase_wallet_quests_optimism_rewards_transfers.sql:42]: Hi @MSilb7, I try to fix the syntax error here, but I may not fully understand the code logics.
Please review this part.

**Comment by Hosuke** [comment]
Try to fix all labels compile-time error in this PR.

**Comment by Hosuke** [comment]
No idea for [the test failure](https://github.com/duneanalytics/spellbook/actions/runs/4422577401/jobs/7755640338#step:10:52):
https://dune.com/queries/2209467?d=1

File: models/coinbase_wallet_quests/optimism/coinbase_wallet_quests_optimism_rewards_transfers.sql (hunk 1 above)
Comment (MSilb7) [reply models/coinbase_wallet_quests/optimism/coinbase_wallet_quests_optimism_rewards_transfers.sql:42]: Hi, yeah seems good. The intent is to say:

- Pull anything in the last week (incremental)
- If this is the first time we're seeing a distributor address, pull everything (i.e. if the distributor existed for > 1 week before the PR gets merged, we're good)

**Review by Hosuke — APPROVED** [review]
I renamed the path name `labels/dex/Persona/*` to lowercase `labels/dex/persona/*`, and this makes this PR seems having lots of file changes.

In fact only 3 models added in this PR by @MSilb7:
1. [coinbase_wallet_quests_optimism_distributor_addresses](https://dune.com/queries/2239436?d=1)
2. [coinbase_wallet_quests_optimism_rewards_transfers](https://dune.com/queries/2239447)
3. [labels_optimism_coinbase_wallet_quest_participants](https://dune.com/queries/2239451)

**Review by jeff-dude — APPROVED** [review]

**Pull request merged**